aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -908,14 +974,22 @@ mod tests {
 
         let mut base_update = BTreeMap::new();
         let mut additional_update = BTreeMap::new();
-        base_update.insert(key.clone(), GroupWrite {
```

Comment:
We test squashing for noops, and empty groups, but not when both have non-empty and different sizes, I think we should add 1-2 tests for that, as this is precisely what exposes the bug.